appshell/appshell_extensions_mac.mm

```diff
@@ -436,17 +483,39 @@ int32 ShowSaveDialog(ExtensionString title,
     }
 
     [savePanel setNameFieldStringValue:[NSString stringWithUTF8String:proposedNewFilename.c_str()]];
-    [savePanel beginSheetModalForWindow:[NSApp mainWindow] completionHandler:nil];
-    
-    if ([savePanel runModal] == NSFileHandlingPanelOKButton)
-    {
-        NSURL* theFile = [savePanel URL];
-        absoluteFilepath = [[theFile path] UTF8String];
 
-    }
-    [NSApp endSheet:savePanel];
-
-    return NO_ERROR;
+    // cache the browser details, so has to access
+    // the same inside the completionHandler block.
+    CefBrowserDetails browserDetails;
+    browserDetails.m_browser  = browser;
+    browserDetails.m_response = response;
+    
+    s_panel_to_browser[savePanel] = browserDetails;
+    
+    [savePanel beginSheetModalForWindow:[NSApp mainWindow] completionHandler: ^(NSInteger returnCode)
+     {
+         panel_iterator _browserDetails;
+         _browserDetails = s_panel_to_browser.find(savePanel);
+         if(_browserDetails != s_panel_to_browser.end()){
+             
+             CefRefPtr<CefProcessMessage> _response = _browserDetails->second.m_response;
```

Comment:
Good catch Abhijith 👍 I will make the change. I would have forgotten to remote that.